### PR TITLE
token tests

### PR DESCRIPTION
**Pull request by alexytsu**
Follow-up to https://github.com/helix-onchain/filecoin/pull/24
- superficial test refactoring
- re-order and condense some unit tests
- make clean removes coverage profiles
- add edge cases and condense tests

### CODE REVIEW

File: fil_fungible_token/src/token/mod.rs [hunk 1]
```diff
@@ -433,337 +439,340 @@ mod test {
...
         let mut token = new_token();
 
-        let treasury = 1;
-        token.mint(TOKEN_ACTOR_ADDRESS, treasury, &TokenAmount::from(1_000_000), &[]).unwrap();
+        token.mint(TOKEN_ACTOR_ADDRESS, TREASURY, &TokenAmount::from(1_000_000), &[]).unwrap();
 
-        let balance = token.balance_of(treasury).unwrap();
-        assert_eq!(balance, TokenAmount::from(1_000_000));
+        // balance and total supply both went up
+        assert_eq!(token.balance_of(TREASURY).unwrap(), TokenAmount::from(1_000_000));
```
Comment (anorth) [inline fil_fungible_token/src/token/mod.rs:445]: Idea: add an `assert_balance(TREASURY, 1_000_000)` helper

File: fil_fungible_token/src/token/mod.rs [hunk 2]
```diff
@@ -433,337 +439,340 @@ mod test {
...
+        assert_eq!(token.total_supply(), TokenAmount::from(1_000_000));
+
+        // mint again to same address
+        token.mint(TOKEN_ACTOR_ADDRESS, TREASURY, &TokenAmount::from(1_000_000), &[]).unwrap();
+        assert_eq!(token.balance_of(TREASURY).unwrap(), TokenAmount::from(2_000_000));
+        assert_eq!(token.total_supply(), TokenAmount::from(2_000_000));
+
+        // mint to a different address
+        token.mint(TOKEN_ACTOR_ADDRESS, ALICE, &TokenAmount::from(1_000_000), &[]).unwrap();
+        assert_eq!(token.balance_of(ALICE).unwrap(), TokenAmount::from(1_000_000));
```
Comment (anorth) [inline fil_fungible_token/src/token/mod.rs:469]: Check TREASURY's balance too

File: fil_fungible_token/src/token/mod.rs [hunk 3]
```diff
@@ -433,337 +439,340 @@ mod test {
...
+        token.transfer(ALICE, ALICE, BOB, &TokenAmount::from(60), "abort".as_bytes()).unwrap_err();
 
-        // balances didn't change
-        let balance = token.balance_of(owner).unwrap();
-        assert_eq!(balance, TokenAmount::from(100));
-        let balance = token.balance_of(receiver).unwrap();
-        assert_eq!(balance, TokenAmount::from(0));
+        // balances unchanged
+        assert_eq!(token.balance_of(ALICE).unwrap(), TokenAmount::from(100));
+        assert_eq!(token.balance_of(BOB).unwrap(), TokenAmount::from(0));
```
Comment (anorth) [inline fil_fungible_token/src/token/mod.rs:608]: Just 'cause I'm paranoid, test transferring non-zero value to self with a receiver hook that aborts.

**Review by anorth — APPROVED** [review]

**Comment by alexytsu** [comment]
I've disabled coverage decrease as a "status" failure for now. I think the next PR should account for this, but something to keep an eye on if it becomes too noisy.
